src/deepObserve.ts

```diff
@@ -83,15 +83,21 @@ export function deepObserve<T = any>(
                 )
                 // update paths
                 for (let i = change.index + change.addedCount; i < change.object.length; i++) {
-                    const entry = entrySet.get(change.object[i])
-                    if (entry) entry.path = "" + i
+                    if (isRecursivelyObservable(change.object[i])) {
+                        const entry = entrySet.get(change.object[i])
+                        if (entry) entry.path = "" + i
+                    }
                 }
                 break
         }
     }
 
+    function isRecursivelyObservable(thing: any) {
```

Comment:
This function can be moved to the top level of the module, as it doesn't depend on anything in the closure